storage/src/meta/mod.rs

```diff
@@ -489,6 +489,9 @@ impl BlobMetaInfo {
                 }
             }
 
+            if last_end > end {
```

Comment:
Should compare `last_end` with `batch_end` ? And `last_end` == `last_end` should be valid

Reply:
let mut last_end = entry.aligned_uncompressed_end();

Reply:
@kevinXYin

Reply:
in while loop already compare last_end with batch_end , for https://github.com/dragonflyoss/image-service/issues/693 issue last chunk's aligned_uncompressed_end not reach the batch_end which is state.uncompressed_size. And I am not sure is this valid (state.uncompressed_size > last chunk's aligned_uncompressed_end) ? But I think just checking the last_end to cover the end is enough.  And , yes last_end == end should be valid , I will send another pr to change to `if last_end >= end` , does this make sense ? @changweige @hsiangkao

Reply:
The root cause of this problem is blob's uncompressed size is wrongly calculated. I have raised a PR to fix the image format. https://github.com/dragonflyoss/image-service/pull/712